comp/core/hostname/service_test.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package hostname
+
+import (
+	"context"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGet(t *testing.T) {
+	t.Cleanup(func() {
+		// erase cache
+		cache.Cache.Delete(cache.BuildAgentKey("hostname"))
+		config.Datadog.Set("hostname", "")
+	})
+	config.Datadog.Set("hostname", "test-hostname")
```

Comment:
Sadly you cannot use MockConfig here.

Reply:
I tried a couple of ways to do this, and I could technically use config.MockModule by doing something like
```go
	s := fxutil.Test[Component](t,
		config.MockModule,
		Module,
		fx.Replace(config.MockParams{
			SetupConfig: true,
			Overrides: map[string]interface{}{
				"hostname": "test-hostname",
			},
		}),
		fx.Invoke(func(_ config.Component) {}),
	)
```
but that seemed like a bad idea, since it's relying on the fact that the mock config sets config.Datadog as a side effect, which may not always be true.

A better solution would be to rework `pkg/util/hostname` to use an injected `Config` instead of the global one, but that would require some pretty substantial changes (among other things, all of the other hostname dependencies like the azure and fargate lookup functions would also need to be componentized).